### PR TITLE
Apply more MVVM approach to Sponsors view

### PR DESCRIPTION
**Pull request by k-kagurazaka**
## Issue
None

## Overview (Required)
This PR has the same purpose as #256 and #304, so diets `SponsorsFragment` and adds tests for `SponsorshipsViewModel`. 
- Use `ObservableList` to bind `Adapter`.
- Almost all logic is moved from `Fragment` to `ViewModel`.
- Merge `AssetsUtil` into `ResourceResolver` for testing.
- Add tests for `SponsorshipsViewModel`.

### CODE REVIEW

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/helper/ResourceResolver.java [hunk 1]
```diff
@@ -1,10 +1,48 @@
...
+
+    public String getString(@StringRes int resId) {
+        return context.getString(resId);
+    }
+
+    public String getString(@StringRes int resId, Object... formatArgs) {
+        return context.getString(resId, formatArgs);
+    }
+
+    public String loadJSONFromAsset(final String jsonFileName) {
```
Comment (k-kagurazaka) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/helper/ResourceResolver.java]: We have another option that split this class into `ResourceResolver` and `AssetsResolver`.
However I think it's little redundant.
What do you think about?

File: app/src/test/java/io/github/droidkaigi/confsched2017/util/RxTestSchedulerRule.kt [hunk 2]
```diff
@@ -5,7 +5,7 @@ import io.reactivex.plugins.RxJavaPlugins
 import io.reactivex.schedulers.TestScheduler
 import org.junit.rules.ExternalResource
 
-class RxTestSchedulerRule : ExternalResource() {
+object RxTestSchedulerRule : ExternalResource() {
```
Comment (k-kagurazaka) [inline app/src/test/java/io/github/droidkaigi/confsched2017/util/RxTestSchedulerRule.kt]: TestScheduler instance must be singleton.
Now `RxTestSchedulerRule` is used from multiple test class, so this must be `object`.

**Comment by konifar** [comment]
:eyes:

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/helper/ResourceResolver.java (hunk 1 above)
Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/helper/ResourceResolver.java]: I agree to add it to `ResourceResolver`.
In fact, we may provide new API for sponsors data. So it's fine like this.

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/helper/ResourceResolver.java [hunk 3]
```diff
@@ -1,10 +1,48 @@
...
+        String json = null;
+        try {
+            InputStream is = context.getAssets().open("json/" + jsonFileName);
+            int size = is.available();
+            byte[] buffer = new byte[size];
+            is.read(buffer);
+            is.close();
+            json = new String(buffer, "UTF-8");
+        } catch (IOException e) {
+            Timber.e(TAG, e.getMessage(), e);
```
Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/helper/ResourceResolver.java]: You can write like this
`Timber.tag(TAG).e(e.getMessage(), e)`

File: app/src/main/java/io/github/droidkaigi/confsched2017/viewmodel/SponsorshipViewModel.java [hunk 4]
```diff
@@ -39,11 +42,11 @@ public String getCategory() {
...
     }
 
-    public List<SponsorViewModel> getSponsors() {
-        return sponsors;
+    public ObservableList<SponsorViewModel> getSponsorViewModels() {
+        return sponsorViewModels;
     }
 
-    private List<SponsorViewModel> setSponsorship(Sponsorship sponsorship) {
+    private List<SponsorViewModel> convertSponsor(Sponsorship sponsorship) {
```
Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched2017/viewmodel/SponsorshipViewModel.java]: Thanks!

**Comment by konifar** [comment]
LGTM! 💯 You improve this project with amazing speed! Thanks! 👏

**Pull request merged**